src/integrations/mailerlite/class-mailerlite.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Mailerlite integration class.
+ *
+ * @package Eightshift_Forms\Integrations
+ */
+
+declare( strict_types=1 );
+
+namespace Eightshift_Forms\Integrations\Mailerlite;
+
+use Eightshift_Forms\Hooks\Filters;
+use Eightshift_Forms\Exception\Missing_Filter_Info_Exception;
+use \MailerLiteApi\MailerLite as MailerLiteClient;
+/**
+ * Mailerlite integration class.
+ */
+class Mailerlite {
+
+  /**
+   * Mailerlite Api client.
+   *
+   * @var MailerLiteClient
+   */
+  private $client;
+
+  /**
+   * Our own implementation of Mailerlite Client.
+   *
+   * @var Mailerlite_Client_Interface
+   */
+  private $mailerlite_client;
+
+  /**
+   * Constructs object
+   *
+   * @param Mailerlite_Client_Interface $mailerlite_client Mailerlite client.
+   */
+  public function __construct( Mailerlite_Client_Interface $mailerlite_client ) {
```

Comment:
I reckon that all these interfaces look the same (just getter and setter).

Interfaces should be abstract on purpose. So it makes no sense to have two same interfaces but with specific names (`Mailerlite_Client_Interface` and `Mailchimp_Marketing_Client_Interface`). Defeats the purpose of using the interface.

It would be better to just have one `ClientInterface` which has the methods that can be implemented in a different way by `Mailerlite` and `Mailchimp` client classes. Then when injecting to classes, the type hint is against a general interface, but the actual implementation will be the classes implementing that interface.

Reply:
Agreed